prow/cmd/tide/main.go

```diff
@@ -159,33 +144,82 @@ func main() {
 	if err != nil {
 		logrus.WithError(err).Fatal("Error constructing mgr.")
 	}
-	c, err := tide.NewController(
-		githubSync,
-		githubStatus,
-		mgr,
-		cfg,
-		git.ClientFactoryFrom(gitClient),
-		o.maxRecordsPerPool,
-		opener,
-		o.historyURI,
-		o.statusURI,
-		nil,
-		o.github.AppPrivateKeyPath != "",
-	)
-	if err != nil {
-		logrus.WithError(err).Fatal("Error creating Tide controller.")
+
+	var c *tide.Controller
+	var gitClient *prowgit.Client
+	if !o.githubOnly && cfg().Tide.Gerrit != nil {
```

Comment:
The behavior doesn't seem to align with the flag description IMHO. The flag description sounds to me like you could run tide for both github and gerrit when in fact you can't and if there is config for both, one of `githubOnly` or `gerritOnly` must be set.

If that is what we end up building, we should  clarify the flags but also add error handling and error if config for both is present and none of the flags is set

Reply:
yeah you're right, my initial plan was one service handling both, but then realize:
- there is no such a use case yet
- it's not really that complicated to run one tide service for each provider

I'll update the flags to make it clear that only one provider is allowed at a time

Reply:
updated, @alvaroaleman please take another look